server/src/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -1350,7 +1364,23 @@ protected boolean isHostInMaintenance(HostVO host, List<VMInstanceVO> runningVms
             return CollectionUtils.isEmpty(failedMigrations) ?
                     setHostIntoMaintenance(host) :
                     setHostIntoErrorInMaintenance(host, failedMigrations);
+        } else if (retryHostMaintenance.containsKey(host.getId())) {
```

Comment:
you do not need the else here. I mean, the other if body will always return. Therefore, you can simply follow with an IF. This will change a bit the formatting.